File: src/main/scala/com/wixpress/embed/mysql/runtime/Mysqld.scala
package com.wixpress.embed.mysql.runtime

import java.io.File

import com.wixpress.embed.mysql.config.MysqldConfig
import de.flapdoodle.embed.process.extract.IExtractedFileSet
import collection.JavaConversions._
/**
 * @author viliusl
 * @since 18/09/14
 */
object Mysqld {

  def getCommandLine(config: MysqldConfig, extractedFiles: IExtractedFileSet, pidFile: File) : java.util.List[String] = {
    val baseDir = s"${extractedFiles.generatedBaseDir()}"

    List[String](
      extractedFiles.executable().getAbsolutePath,
      s"--basedir=$baseDir",
      s"--datadir=$baseDir/data",
      s"--plugin-dir=$baseDir/lib/plugin",
      s"--pid-file=${pidFile.getAbsolutePath}",
      s"--port=${config.port}"
    )
  }

}

File: src/main/scala/com/wixpress/embed/mysql/runtime/Mysqld.java
package com.wixpress.embed.mysql.runtime;

import java.io.File;
import java.util.List;

import com.wixpress.embed.mysql.config.MysqldConfig;
import de.flapdoodle.embed.process.collections.Collections;
import de.flapdoodle.embed.process.extract.IExtractedFileSet;

/**
 * @author viliusl
 * @since 18/09/14
 */
public class Mysqld {

    public static List<String> getCommandLine(
            final MysqldConfig config,
            final IExtractedFileSet extractedFiles,
            final File pidFile) {

        final String baseDir = extractedFiles.generatedBaseDir().getAbsolutePath();

        return Collections.newArrayList(
            extractedFiles.executable().getAbsolutePath(),
            String.format("--basedir=%s", baseDir),
            String.format("--datadir==%s/data", baseDir),
            String.format("--plugin-dir==%s/lib/plugin", baseDir),
            String.format("--pid-file==%s", pidFile.getAbsolutePath()),
            String.format("--port==%s", config.port()),
            "--console");//windows specific
    }

}
